clients/cloudsourceapi/domain.go

```diff
@@ -0,0 +1,67 @@
+package cloudsourceapi
+
+import (
+	"fmt"
+	"strings"
+	"time"
+)
+
+// AccessToken represents a token to use for api requests
+type AccessToken struct {
+	AccessToken  string    `json:"access_token"`
+	Expiry       time.Time `json:"expiry"`
+	RefreshToken string    `json:"refresh_token"`
+	TokenType    string    `json:"token_type"`
+}
+
+// PubSubNotification represents a PubSub message coming from Cloud Source Repo
+type PubSubNotification struct {
+	Name           string          `json:"name"`
+	Url            string          `json:"url"`
+	EventTime      time.Time       `json:"eventTime"`
+	RefUpdateEvent *RefUpdateEvent `json:"refUpdateEvent,omitempty"`
+}
+
+// RefUpdateEvent represents the Update event from Cloud Source Repo
+type RefUpdateEvent struct {
+	Email      string               `json:"email"`
+	RefUpdates map[string]RefUpdate `json:"refUpdates"`
+}
+
+// RefUpdate represents the Update for each reference from a Cloud Source Repo
+type RefUpdate struct {
+	RefName    string `json:"refName"`
+	UpdateType string `json:"updateType"`
+	OldId      string `json:"oldId"`
+	NewId      string `json:"newId"`
+}
+
+// GetRepoSource returns the repository source
+func (psn *PubSubNotification) GetRepoSource() string {
+	return "source.developers.google.com"
```

Comment:
Besides being used to clone the repo this is also what will show up in the web interface. Once we see those we'll have to check how well things render, given that it's pretty long.

Reply:
For the interface would be better to use the link which goes to the UI `source.cloud.google.com`.
But I am not sure what would be the best way to do this.

Reply:
Yes, weird that they have both `source.cloud.google.com` and `source.developer.google.com`. Can they make up their mind? ;)